ruby_on_rails/rspec.md

```diff
@@ -112,21 +112,43 @@ Check that you see `1+2=3` in each app.
 
 ## Optional: catch javascript errors
 
-If you want to catch Javascript errors in your system tests, you can add the following to the `rails_helper.rb`:
+If you want to catch Javascript errors in your system tests, you can create `support/javascript_error_collector.rb`:
 
 ```ruby
-config.after(:each, type: :system, js: true) do
-  errors = page.driver.browser.manage.logs.get(:browser)
-  if errors.present?
+module JavaScriptErrorCollector
+  def collect_errors
+    errors = page.driver.browser.manage.logs.get(:browser)
+
     aggregate_failures 'javascript errors' do
       errors.each do |error|
         expect(error.level).not_to eq('SEVERE'), error.message
+
         next unless error.level == 'WARNING'
+        warn "\e[33m\nJAVASCRIPT WARNING\n#{error.message}\e[0m" if Rails.application.config.show_javascript_warning
 
-        warn 'WARN: javascript warning'
-        warn error.message
+        next unless Rails.application.config.fail_on_javascript_warning
+        expect(error.level).not_to eq('WARNING'), error.message
       end
     end
   end
 end
 ```
+
+Then include it in the `rails_helper.rb`:
+
+```ruby
+config.include JavaScriptErrorCollector, type: :system
+config.after(:each, type: :system, js: true) { collect_errors }
```

Comment:
can we make it so, that it automatically collects errors for `js: true` as soon as we include the module? so we would have only `include` in the `rails_helper.rb`.
Also, wouldn't `JavascriptErrorReporter` be a bit better name? It doesn't collect it anywhere, it reports it on the console